crates/core/src/protocols/evalcheck/prove.rs

```diff
@@ -222,28 +223,36 @@ where
 
 		let projected_mles = calculate_projected_mles(
 			&projected_bivariate_metas,
-			&mut self.memoized_queries,
+			&mut self.memoized_data,
 			&self.projected_bivariate_claims,
 			self.witness_index,
 			self.backend,
 		)?;
 
 		fill_eq_witness_for_composites(
 			&projected_bivariate_metas,
-			&mut self.memoized_queries,
+			&mut self.memoized_data,
 			&self.projected_bivariate_claims,
 			self.witness_index,
 			self.backend,
 		)?;
 
 		for (claim, meta, projected) in izip!(
 			std::mem::take(&mut self.projected_bivariate_claims),
-			projected_bivariate_metas,
+			&projected_bivariate_metas,
 			projected_mles
 		) {
 			self.process_sumcheck(claim, meta, projected)?;
 		}
 
+		memoize_partial_evals(
```

Comment:
Why is this called here? The results don't get used in this method AFAICT?

Reply:
This method builds a map from (id, eval_point) to MultilinearWitness, which will be used in ring_switch instead of storing evals separately

Reply:
Then this code should be in `ring_switch` if it's only used in `ring_switch`

Reply:
Now it's part of MemoizedData, so I think it's the right place